src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,51 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
+
+  getDamage(damage) {
+    this.health -= damage;
+
+    if (this.health <= 0) {
+      this.die();
+    }
+  }
+
+  die() {
+    Animal.alive = Animal.alive.filter((animal) => animal !== this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+  }
+
+  bite(target) {
+    if (target instanceof Herbivore) {
+      if (!target.hidden) {
+        target.getDamage(50);
+      }
+    }
```

Comment:
[CODE PERFOMANCE] - After the `target.getDamage(50);` call, if the health of the target becomes less than or equal to 0, you should remove the dead animal from the `Animal.alive` array. This can be done by filtering the array to exclude animals with health less than or equal to 0. Avoid using nested loops or indexOf + splice for performance reasons.